### ISSUE TITLE
Input Validation Status not conveyed via Screen reader

### ISSUE THREAD

**Issue by jake-costa** [jpmorganchase/salt-ds#4736]
### Latest version

- [x] I have tested the latest version

### Description

Input Validation Status not conveyed via Screen reader. [WCAG 4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html)

Error status should have aria-invalid
For warning, and success. We could use aria-describedby to target the aria--label of the icon or use another method to add hidden accessible text to convey the status.

Also, to support [WCAG 4.1.3 Status Messages](https://www.w3.org/WAI/WCAG22/Understanding/status-messages.html), After navigating away from an empty required form field for collecting the user's first name, an error icon is added to the form field. The screen reader should announce "error: first name required." for example.

### Steps to reproduce

Using a screen reader, navigate to an input field that contains an error, warning, or success. notice that the status is not conveyed. 

### Package name(s)

Core (@salt-ds/core)

### Browser

- [x] Chrome
- [x] Safari
- [x] Firefox
- [ ] Microsoft Edge

### Operating system

- [x] macOS
- [x] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [x] I am an employee of JPMorgan Chase & Co.

**Comment by bhoppers2008** [comment]
Migrated to Jira.
